crates/crochet_infer/src/infer_expr.rs

```diff
@@ -479,12 +487,8 @@ pub fn infer_expr(ctx: &mut Context, expr: &mut Expr) -> Result<(Subst, Type), S
             }
 
             let s = compose_many_subs(&ss);
-            let t = Type {
-                kind: TypeKind::Tuple(ts),
-                provenance: Some(Box::from(expr.to_owned())),
```

Comment:
We set the provenance at the bottom of the function now so that it's done for all expression kinds.